libstorage/BinLogHandler.cpp

```diff
@@ -394,21 +413,39 @@ uint32_t BinLogHandler::decodeEntries(const bytes& buffer, uint32_t& offset,
         /*BINLOG_HANDLER_LOG(TRACE) << LOG_DESC("entry info") << LOG_KV("idx in entries", idx)
                                   << LOG_KV("_id_", entry->getID())
                                   << LOG_KV("_status_", (uint32_t)entry->getStatus());*/
+        int flagSize = (vecField.size() - 3) / 8 + 1;  // exclude STATUS,NUM_FIELD,ID_FIELD
+        uint8_t* usedFlag = new uint8_t[flagSize];     // bit-based used to represent a field
+        int fieldIdx = 0;                              // field index
+        for (int i = 0; i < flagSize; i++)
```

Comment:
```suggestion
       uint8_t * usedFlag = buffer+ offset;
       offset+= flagSize;
```

Reply:
Done

Reply:
use array as a unit to process, easy to understand